### PR TITLE
[docker] Publish installed packages in docker containers

### PR DESCRIPTION
**Pull request by krfricke**
## Why are these changes needed?

This PR adds information about the installed packages in our Docker images to the Ray docs.

https://anyscale-ray--35390.com.readthedocs.build/en/35390/ray-overview/installation.html#docker-source-images

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/ray-overview/installation.rst [hunk 1]
```diff
@@ -495,3 +495,27 @@ that you've cloned the git repository.
 .. code-block:: bash
 
   python -m pytest -v python/ray/tests/test_mini.py
+
+
+Installed Python dependencies
```
Comment (richardliaw) [inline doc/source/ray-overview/installation.rst:500]: How frequently should this be updated? Should the release manager be updating this?

Comment (krfricke) [reply doc/source/ray-overview/installation.rst:500]: Yes, I think the release manager should update it while preparing a release (for the `latest` documentation). I can add this to the release process doc.

**Review by richardliaw — APPROVED** [review]
(approval conditional on release process)

**Comment by krfricke** [comment]
(added to release docs)

**Pull request merged**